### PR TITLE
`azurerm_postgresql_flexible_server`: fix the logic in update

### PR DESCRIPTION
**Pull request by ziyeqf**
fix https://github.com/hashicorp/terraform-provider-azurerm/issues/20900

the root cause is: `administrator_login` is only able to be set when `authentication.password_auth_enabled` set to `true`...

`completeUpdate` also failed on main branch, may fix it in another PR.
Screenshot 2023-03-14 at 17 19 07

### CODE REVIEW

**Comment by nbjohnson** [comment]
I just want to clarify, is this change going to allow me to remove password auth after creation? Or is this just going to clarify the error output that once password auth is enabled it cannot be disabled?

**Comment by ziyeqf** [comment]
> I just want to clarify, is this change going to allow me to remove password auth after creation? Or is this just going to clarify the error output that once password auth is enabled it cannot be disabled?

It's going to allow update `password_auth_enabled` from `true` to `false`, which get an error before. And when doing so, there is no need to remove `administrator_login` and `administrator_password`. it allows to do the following steps:

1. first time apply
```
resource "azurerm_postgresql_flexible_server" "test" {
# other properties...
  administrator_login    = "adminZhenTerraform"
  administrator_password = "QAZwsx123"

  authentication {
    password_auth_enabled         = true
  }
}
```

2. update to disable `password_auth_enabled`
```
resource "azurerm_postgresql_flexible_server" "test" {
# other properties...
  administrator_login    = "adminZhenTerraform"
  administrator_password = "QAZwsx123"

  authentication {
    password_auth_enabled         = false
  }
}
```
```

**Comment by ziyeqf** [comment]
> @ziyeqf Ok thanks for the clarification. I guess I don't care if it allows users to pass in a login and password that it will just ignore. If you look at my code I was passing in `null` for both of them while at the same time disabling password auth. So my problem was that I just couldn't turn off password auth as it still thought login and password were present even when I was passing in `null`.
> 
> 
> 
> ```
> 
> administrator_login = var.password_auth_enabled == true ? var.administrator_login : null
> 
> administrator_password = var.password_auth_enabled == true ? var.administrator_password : null
> 
> ```

This PR will fix the bug which blocks disabling password auth. And my personal suggestion is not to pass "administrator_login" a "null" value, because once it has been set,setting it to another value (except "null") will force a new server to be created.

**Comment by nbjohnson** [comment]
@ziyeqf Thanks for putting this together as having the ability to disable password auth after creation is a big requirement for me so I can switch existing servers from password auth to only AD auth. Do we know when this change might be implemented as it is blocking my implementation.

File: internal/services/postgres/postgresql_flexible_server_resource.go [hunk 1]
```diff
@@ -387,7 +387,7 @@ func resourcePostgresqlFlexibleServerCreate(d *pluginsdk.ResourceData, meta inte
 				return fmt.Errorf("`administrator_password` is required when `create_mode` is `Default` and `authentication.password_auth_enabled` is set to `true`")
 			}
 		} else if adminLoginSet || adminPwdSet {
-			return fmt.Errorf("`administrator_login` and `administrator_password` cannot be set when `authentication.password_auth_enabled` is set to `false`")
+			return fmt.Errorf("`administrator_login` and `administrator_password` cannot be set in creation when `authentication.password_auth_enabled` is set to `false`")
```
Comment (katbyte) [inline internal/services/postgres/postgresql_flexible_server_resource.go:390]: grammer
```suggestion
			return fmt.Errorf("`administrator_login` and `administrator_password` cannot be set during creation when `authentication.password_auth_enabled` is set to `false`")
```

File: internal/services/postgres/postgresql_flexible_server_resource.go [hunk 2]
```diff
@@ -629,12 +629,13 @@ func resourcePostgresqlFlexibleServerUpdate(d *pluginsdk.ResourceData, meta inte
...
 				return fmt.Errorf("`administrator_password` is required when `authentication.password_auth_enabled` is set to `true`")
 			}
-		} else if adminLoginSet || adminPwdSet {
-			return fmt.Errorf("`administrator_login` and `administrator_password` cannot be set when `authentication.password_auth_enabled` is set to `false`")
 		}
 
 		if d.HasChange("administrator_login") {
 			requireUpdateOnLogin = true
+			if adminLoginSet && !pwdEnabled {
+				return fmt.Errorf("when `administrator_login` is set first time, `authentication.password_auth_enabled` must be set to `true`")
```
Comment (katbyte) [inline internal/services/postgres/postgresql_flexible_server_resource.go:637]: grammer
```suggestion
				return fmt.Errorf("when `administrator_login` is first set, `authentication.password_auth_enabled` must be set to `true`")
```

File: website/docs/r/postgresql_flexible_server.html.markdown [hunk 3]
```diff
@@ -92,6 +92,8 @@ The following arguments are supported:
 
 -> **Note:** Once `administrator_login` is specified, changing this forces a new PostgreSQL Flexible Server to be created.
 
+-> **Note:** To create with `administrator_login` specified or update with it specified first time, `authentication.password_auth_enabled` must be set to `true`.
```
Comment (katbyte) [inline website/docs/r/postgresql_flexible_server.html.markdown:95]: as above

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by nbjohnson** [comment]
Any update on this fix? I need this fix as soon as possible as it is blocking my implementation

**Review by katbyte — APPROVED** [review]
LGTM 🦀

**Pull request merged**